plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,403 @@
+            for (String field : documentFields) {
+                getMappedModelInputFromDocuments(ingestDocument, modelParameters, field, field);
+            }
+
+        } else {
+            Map<String, String> inputMapping = processInputMap.get(inputMapIndex);
+            for (Map.Entry<String, String> entry : inputMapping.entrySet()) {
+                // model field as value, document field as key
+                String originalFieldName = entry.getKey();
+                String modelInputFieldName = entry.getValue();
+                getMappedModelInputFromDocuments(ingestDocument, modelParameters, originalFieldName, modelInputFieldName);
+            }
+        }
+
+        ActionRequest request = getRemoteModelInferenceRequest(modelParameters, inferenceProcessorAttributes.getModelId());
+
+        client.execute(MLPredictionTaskAction.INSTANCE, request, new ActionListener<>() {
+
+            @Override
+            public void onResponse(MLTaskResponse mlTaskResponse) {
+                ModelTensorOutput modelTensorOutput = (ModelTensorOutput) mlTaskResponse.getOutput();
+                if (processOutputMap == null || processOutputMap.isEmpty()) {
+                    appendFieldValue(modelTensorOutput, null, DEFAULT_OUTPUT_FIELD_NAME, ingestDocument);
+                } else {
+                    // outMapping serves as a filter to modelTensorOutput, the fields that are not specified
+                    // in the outputMapping will not write to document
+                    Map<String, String> outputMapping = processOutputMap.get(inputMapIndex);
+
+                    for (Map.Entry<String, String> entry : outputMapping.entrySet()) {
+                        String originalModelOutputFieldName = entry.getValue();
+                        String newDocumentFieldName = entry.getKey();
+                        appendFieldValue(modelTensorOutput, originalModelOutputFieldName, newDocumentFieldName, ingestDocument);
+                    }
+                }
+                batchPredictionListener.onResponse(null);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                batchPredictionListener.onFailure(e);
+            }
+        });
+
+    }
+
+    private void getMappedModelInputFromDocuments(
+        IngestDocument ingestDocument,
+        Map<String, String> modelParameters,
+        String originalFieldName,
+        String modelInputFieldName
+    ) {
+        String originalFieldPath = getFieldPath(ingestDocument, originalFieldName);
+        if (originalFieldPath != null) {
+            Object originalFieldValue = ingestDocument.getFieldValue(originalFieldPath, Object.class);
+            String originalFieldValueAsString = toString(originalFieldValue);
+            updateModelParameters(modelInputFieldName, originalFieldValueAsString, modelParameters);
+        }
+        // check for nested array
+        else {
+            if (originalFieldName.contains(DOT_SYMBOL)) {
```

Comment:
Add `else` for this if

Reply:
added else statement

Reply:
https://github.com/opensearch-project/ml-commons/blob/84abae78369b943e634cc22039667ef24c173939/plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java#L250

https://github.com/opensearch-project/ml-commons/blob/84abae78369b943e634cc22039667ef24c173939/plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java#L255